Libraries/LibWeb/HTML/HTMLMediaElement.cpp

```diff
@@ -813,138 +813,140 @@ WebIDL::ExceptionOr<void> HTMLMediaElement::select_resource()
-            HTML::main_thread_event_loop().spin_until(GC::create_function(heap(), [&]() { return ran_media_element_task; }));
-        };
-
-        // 1. ⌛ If the src attribute's value is the empty string, then end the synchronous section, and jump down to the failed with attribute step below.
-        auto source = get_attribute_value(HTML::AttributeNames::src);
-        if (source.is_empty()) {
-            failed_with_attribute("The 'src' attribute is empty"_string);
-            return {};
-        }
+                    ran_media_element_task = true;
+                });
 
-        // 2. ⌛ Let urlString and urlRecord be the resulting URL string and the resulting URL record, respectively, that would have resulted from parsing
-        //    the URL specified by the src attribute's value relative to the media element's node document when the src attribute was last changed.
-        auto url_record = document().parse_url(source);
+                // 7. Wait for the task queued by the previous step to have executed.
+                HTML::main_thread_event_loop().spin_until(GC::create_function(heap(), [&]() { return ran_media_element_task; }));
+            };
 
-        // 3. ⌛ If urlString was obtained successfully, set the currentSrc attribute to urlString.
-        if (url_record.has_value())
-            m_current_src = url_record->to_string();
+            // 1. ⌛ If the src attribute's value is the empty string, then end the synchronous section, and jump down to the failed with attribute step below.
+            auto source = get_attribute_value(HTML::AttributeNames::src);
+            if (source.is_empty()) {
+                failed_with_attribute("The 'src' attribute is empty"_string);
+                return;
+            }
 
-        // 4. End the synchronous section, continuing the remaining steps in parallel.
+            // 2. ⌛ Let urlString and urlRecord be the resulting URL string and the resulting URL record, respectively, that would have resulted from parsing
+            //    the URL specified by the src attribute's value relative to the media element's node document when the src attribute was last changed.
+            auto url_record = document().parse_url(source);
 
-        // 5. If urlRecord was obtained successfully, run the resource fetch algorithm with urlRecord. If that algorithm returns without aborting this one,
-        //    then the load failed.
-        if (url_record.has_value()) {
-            TRY(fetch_resource(*url_record, move(failed_with_attribute)));
-            return {};
-        }
+            // 3. ⌛ If urlString was obtained successfully, set the currentSrc attribute to urlString.
+            if (url_record.has_value())
+                m_current_src = url_record->to_string();
 
-        failed_with_attribute("Failed to parse 'src' attribute as a URL"_string);
+            // 4. End the synchronous section, continuing the remaining steps in parallel.
```

Comment:
Apologies for the big delay in review, have been busy lately. But I believe if I am reading the specification correctly, step 5 onwards should not be executed in the context on the microtask. Instead, we should run it in parallel. This is currently achived by invoking: `Platform::EventLoopPlugin::the().deferred_invoke`.

I think this is problably quite important considering there is another spec step which blocks until this is complete. This seems quite tricky in general!